pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -366,9 +366,32 @@ func (c *Cluster) checkHealth(ctx context.Context) error {
 		}
 	}
 
+	// failover a mon if its IP is required to be updated due to change in hostNetwork settings
+	monName, updateMonIP := c.isMonIPUpdateRequired()
+	if updateMonIP {
+		c.failMon(len(c.ClusterInfo.Monitors), desiredMonCount, monName)
+	}
+
 	return nil
 }
 
+// verify if the hostnetwork setting was changed on a running cluster by comparing the
+// mon IP and the node IP.
+func (c *Cluster) isMonIPUpdateRequired() (string, bool) {
+	for _, mon := range c.ClusterInfo.Monitors {
+		if c.mapping != nil {
+			monNodeAddress := c.mapping.Schedule[mon.Name].Address
```

Comment:
If host networking is not enabled, is this `Address` field populated? I'm not sure if it is.